cmd/yggdrasil/main.go

```diff
@@ -366,6 +373,23 @@ func run(args yggArgs, ctx context.Context, done chan struct{}) {
 	address := n.core.Address()
 	subnet := n.core.Subnet()
 	public := n.core.GetSelf().Key
+	// Lower permissions from root to something else, if the user wants to
+	if syscall.Getuid() == 0 {
+		if args.rungid > 0 {
+			logger.Infoln("Setting gid to:", args.rungid)
+			if err := setgid(args.rungid); err != nil {
+				logger.Errorln("Failed to set gid:", err)
+				return
```

Comment:
instead of simply returning, you likely want to abort() here.

Not being able to change gid/uid (or egid/euid) is not a failure you should ignore and then simply continue running.